datafusion/physical-expr/src/aggregate/first_last.rs

```diff
@@ -259,8 +295,20 @@ impl FirstValueAccumulator {
                 options: Some(req.options),
             })
             .collect::<Vec<_>>();
-        let indices = lexsort_to_indices(&sort_columns, Some(1))?;
-        Ok((!indices.is_empty()).then_some(indices.value(0) as _))
+
+        if self.is_ignore_null {
+            let indices = lexsort_to_indices(&sort_columns, Some(value.len()))?;
```

Comment:
instead you can use 
```suggestion
            let indices = lexsort_to_indices(&sort_columns, None)?;
```
to be more explicit about we are sorting whole array

Reply:
Fixed. Thanks!